File: dart/euler1.dart
// Euler1 in Dart

int euler1(int size) {
   var retval = 0;

   for(int i=1; i<size; i++)
     if (i%3==0 || i%5==0)
            retval += i;

   return retval ;
}

main() {
    print(euler1(1000));
}

File: c++/euler1.cpp
// Euler1 in C++
#include <iostream>

class Euler1 {
    private:
        int size;
        int result;

        bool isMultipleOfThree(int n) {
            return n % 3 == 0;
        }

        bool isMultipleOfFive(int n) {
            return n % 5 == 0;
        }

    public:
        Euler1(int size) {
            this->size = size;
            this->result = 0;
        }

        void solve() {
            for (int i = 0; i < size; i++) {
                if (isMultipleOfThree(i) || isMultipleOfFive(i)) {
                    result += i;
                }
            }
        }

        int getResult() {
            return result;
        }
};

int main () {
    Euler1 euler1(1000);

    euler1.solve();

    std::cout << "Euler1 = " << euler1.getResult();
    return 0;
}